packages/wrangler/src/dev/miniflare.ts

```diff
@@ -523,6 +526,50 @@ export function buildMiniflareBindingOptions(config: MiniflareBindingsConfig): {
 		}
 	}
 
+	const tails: NonNullable<WorkerOptions["tails"]> = [];
+	const notFoundTails = new Set<string>();
+	for (const tail of config.tails ?? []) {
+		if (tail.service === config.name || config.workerDefinitions === null) {
+			// If this is a tail binding to the current worker or the registry is disabled,
+			// don't bother using the dev registry to look up the address, just bind to it directly.
+			tails.push({ name: tail.service });
+
+			continue;
+		}
+
+		const target = config.workerDefinitions?.[tail.service];
+
+		// Tail consumers are always on the default entrypoint
+		const defaultEntrypoint = target?.entrypointAddresses?.["default"];
+		if (
+			target?.host === undefined ||
+			target.port === undefined ||
+			defaultEntrypoint === undefined
+		) {
+			notFoundTails.add(tail.service);
+		} else {
+			const style = HttpOptions_Style.PROXY;
+			const address = `${defaultEntrypoint.host}:${defaultEntrypoint.port}`;
+
+			tails.push({
+				external: {
+					address,
+					http: {
+						style,
+						cfBlobHeader: CoreHeaders.CF_BLOB,
+					},
+				},
+			});
+		}
+	}
+
+	if (notFoundTails.size > 0) {
+		logger.debug(
```

Comment:
If I am a user and have configured tail consumers that for some reason we couldn't connect to, I'd want that to be surfaced to me. Unless I'm over-indexing on the importance of this?

Reply:
This is for the dev flow, and so I didn't want to add too much information for users. I went with the same messaging structure as service bindings, and tails show a red "not connected" message, which I think is enough?

![Screenshot 2025-04-28 at 14 01 20](https://github.com/user-attachments/assets/943c2856-7e32-4aa7-b579-97d20076fe0e)

Reply:
gotcha! yeah that's perfect!